examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+        f"`--model_path` are provided, then the script is run with `--{default_model_parameter}={default_model}`.",
+    )
+    parser.add_argument(
+        "--max_seq_length",
+        "-L",
+        type=int,
+        default=64,
+        help="Length of segments into which queries are split. `--max_seq_length` includes [CLS] and [SEP] tokens.",
+    )
+    parser.add_argument(
+        "--step",
+        "-s",
+        type=int,
+        default=8,
+        help="Relative shift of consequent segments into which long queries are split. Long queries are split into "
+        "segments which can overlap. Parameter `step` controls such overlapping. Imagine that queries are "
+        "tokenized into characters, `max_seq_length=5`, and `step=2`. In such a case query 'hello' is tokenized "
+        "into segments `[['[CLS]', 'h', 'e', 'l', '[SEP]'], ['[CLS]', 'l', 'l', 'o', '[SEP]']]`.",
+    )
+    parser.add_argument(
+        "--margin",
+        "-g",
+        type=int,
+        default=16,
+        help="A number of subtokens in the beginning and the end of segments which output probabilities are not used "
+        "for prediction computation. The first segment does not have left margin and the last segment does not have "
+        "right margin. For example, if input sequence is tokenized into characters, `max_seq_length=5`, `step=1`, "
+        "and `margin=1`, then query 'hello' will be tokenized into segments `[['[CLS]', 'h', 'e', 'l', '[SEP]'], "
+        "['[CLS]', 'e', 'l', 'l', '[SEP]'], ['[CLS]', 'l', 'l', 'o', '[SEP]']]`. These segments are passed to the "
+        "model. Before final predictions computation, margins are removed. In the next list, subtokens which logits "
+        "are not used for final predictions computation are marked with asterisk: `[['[CLS]'*, 'h', 'e', 'l'*, "
+        "'[SEP]'*], ['[CLS]'*, 'e'*, 'l', 'l'*, '[SEP]'*], ['[CLS]'*, 'l'*, 'l', 'o', '[SEP]'*]]`.",
+    )
+    parser.add_argument(
+        "--batch_size", "-b", type=int, default=128, help="Number of segments which are processed simultaneously.",
+    )
+    args = parser.parse_args()
+    if args.input_manifest is None and args.output_manifest is not None:
+        parser.error("--output_manifest requires --input_manifest")
+    if args.pretrained_name is None and args.model_path is None:
+        setattr(args, default_model_parameter, default_model)
+    for name in ["input_manifest", "input_text", "output_manifest", "output_text", "model_path"]:
+        if getattr(args, name) is not None:
+            setattr(args, name, getattr(args, name).expanduser())
+    return args
+
+
+def load_manifest(manifest: Path):
+    result = []
+    with manifest.open() as f:
+        for i, line in enumerate(f):
+            data = json.loads(line)
+            result.append(data)
+    return result
+
+
+def _get_subtokens_and_subtokens_mask(query: str, tokenizer: TokenizerSpec) -> Tuple[List[str], List[int]]:
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification

Reply:
moved to dataset file